### PR TITLE
Menu latéral

### LINKED ISSUE #429: Sur la page détail d'une API, transformer la navigation horizontale en navigation verticale
- pas d'effet sur scroll
- pas de petite bulle bleu sur les élements de menu
- uniquement des hyper lien avec des ancres

### PR DESCRIPTION
**Pull request by tmerlier**
side_menu

Fix #429

### CODE REVIEW

**Comment by tmerlier** [comment]
> can we keep the menu on top of the screen instead of in the middle of it ?
> 
> image

Sorry, I forgot to push my last commit…

**Comment by rdubigny** [comment]
@tmerlier ok.

Mind my comment about the blue line

**Comment by rdubigny** [comment]
@tmerlier also, I think there is a little glitch which may be easy to fix. When I click on "Supervision", "Limite d'usage" is highlighted instead of "Supervision". 

image

**Comment by rdubigny** [comment]
@tmerlier can you run prettier on this code ?

File: components/api/scrollable-menu/scrollable-section.js [hunk 1]
```diff
@@ -1,10 +1,18 @@
-import React from "react";
+import React, { useRef, useEffect } from "react";
 import PropTypes from "prop-types";
 import Link from "next/link";
 
-const Section = ({ id, title, children }) => {
+const ScrollableSection = ({ id, title, addRef, children }) => {
```
Comment (rdubigny) [inline components/api/scrollable-menu/scrollable-section.js]: this interface is KO for this component : the complexity of addRef refers to internal behaviour and should not get appear outside of the component. A developer willing to use a Section does want to specify id, title and content. He does not want to specify addRef => this kind of complexity should be hidden.

File: pages/api.js [hunk 2]
```diff
@@ -85,40 +87,53 @@ const API = ({ api, services }) => {
...
-
-        <TechnicalDocumentation
-          doc_tech_description={doc_tech_description}
-          doc_tech_link={doc_tech_link}
-          doc_tech_external={doc_tech_external}
-        />
-
-        <Services services={services} />
+        <ScrollableMenu detail={detail}>
+          <ScrollableMenuContext.Consumer>
```
Comment (rdubigny) [inline pages/api.js]: the page body should not appear has a children of the menu : 

- it is semantically incorrect
- it is a pitfall for rendering performance

**Pull request closed without merging**